executor/cook/executor.py

```diff
@@ -456,6 +456,7 @@ def reregistered(self, driver, agent_info):
     def disconnected(self, driver):
         logging.info('Mesos requested executor to disconnect')
         self.disconnect_signal.set()
+        self.stop_signal.set()
```

Comment:
This looks right to me. Based on [the documentation for executor disconnection](http://mesos.apache.org/documentation/latest/executor-http-api/#disconnections): <i>"If framework checkpointing is disabled, the executor is not supposed to retry subscription and gracefully exit."</i>